Speech/tests/Unit/V1/SpeechClientTest.php

```diff
@@ -0,0 +1,167 @@
+        $recognitionConfig = new RecognitionConfig();
+        $recognitionConfig->setEncoding($encoding);
+        $recognitionConfig->setSampleRateHertz($sampleRateHertz);
+        $recognitionConfig->setLanguageCode($languageCode);
+        return $recognitionConfig;
+    }
+
+    /**
+     * @dataProvider createRecognitionAudioDataProvider
+     */
+    public function testCreateRecognitionAudio($audio, $expectedRequestMessage)
+    {
+        $actualRequestMessage = $this->client->createRecognitionAudio($audio);
+        $this->assertEquals($expectedRequestMessage->serializeToJsonString(), $actualRequestMessage->serializeToJsonString());
+    }
+
+    public function createRecognitionAudioDataProvider()
+    {
+        $uri = 'gs://my-bucket/my-audio.flac';
+        $data = 'abcdefgh';
+        $resourceData = 'zyxwvuts';
+        $resource = $this->createResource($resourceData);
+        $recognitionAudio = (new RecognitionAudio())
+            ->setContent('directRequestData');
+        return [
+            [$uri, (new RecognitionAudio())
+                    ->setUri($uri)],
+            [$data, (new RecognitionAudio())
+                    ->setContent($data)],
+            [$resource, (new RecognitionAudio())
+                    ->setContent($resourceData)],
+            [$recognitionAudio, $recognitionAudio]
+        ];
+    }
+
+    /**
+     * @dataProvider recognizeAudioStreamData
+     */
+    public function testRecognizeAudioStream($audio, $expectedContent)
+    {
+        $recognitionConfig = $this->createRecognitionConfig();
+        $config = new StreamingRecognitionConfig();
+        $config->setConfig($recognitionConfig);
+
+        $expectedResponseStream = [
+            new StreamingRecognizeResponse(),
+            new StreamingRecognizeResponse(),
+        ];
+
+        $mockBidiStreamingCall = new MockBidiStreamingCall($expectedResponseStream);
+
+        $this->transport->startBidiStreamingCall(Argument::allOf(
+            Argument::type(Call::class),
+            Argument::which('getMethod', 'google.cloud.speech.v1.Speech/StreamingRecognize')
+        ),
+            Argument::type('array')
+        )
+            ->shouldBeCalledTimes(1)
+            ->willReturn(new BidiStream($mockBidiStreamingCall));
+
+        $responseStream = $this->client->recognizeAudioStream($config, $audio);
+
+        $this->assertSame($expectedResponseStream, iterator_to_array($responseStream));
+
+        /** @var StreamingRecognizeRequest[] $receivedCalls */
+        $receivedCalls = $mockBidiStreamingCall->popReceivedCalls();
+
+        $expectedConfigMessage = new StreamingRecognizeRequest();
+        $expectedConfigMessage->setStreamingConfig($config);
+
+        // Expect one extra call, for the config message
+        $this->assertSame(count($expectedContent) + 1, count($receivedCalls));
+        $this->assertEquals($expectedConfigMessage, $receivedCalls[0]);
+        for ($i = 0; $i < count($expectedContent); $i++) {
```

Comment:
This may just be preferential, so feel free to disregard if you don't see the value -  but this may be easier to reason about if we just popped the first received call off as the "config". Then we wouldn't need to worry about tracking `$i + 1` while asserting on  the other received calls.

Reply:
I like that, fixed.